### ISSUE TITLE
setting related to projections

### ISSUE THREAD

**Issue by arunmarathe** [ClickHouse/ClickHouse#39844]
I was playing with projections. I set the "allow_experimental_projection_optimizatino" setting to 1.
Why does the following show "Obsolete setting. does nothing."?

Should I not be setting it to 1? Is this a bug?

Thanks,
Arun

```arunm-testvm :) select *select * from system.settings where name like '%experimental_projection%';

SELECT *
FROM system.settings
WHERE name LIKE '%experimental_projection%'

Query id: 89a96daf-61a4-4ab6-ac46-4e80ad583df7

┌─name───────────────────────────────────────┬─value─┬─changed─┬─description─────────────────────┬─min──┬─max──┬─readonly─┬─type─┐
│ allow_experimental_projection_optimization │ 1     │       1 │ Obsolete setting, does nothing. │ ᴺᵁᴸᴸ │ ᴺᵁᴸᴸ │        0 │ Bool │
└────────────────────────────────────────────┴───────┴─────────┴─────────────────────────────────┴──────┴──────┴──────────┴──────┘

1 row in set. Elapsed: 0.003 sec.

```

**Comment by arunmarathe** [comment]
Also, I will ask a related question.
How can I tell whether a projection was used by a query?
Can I print explain plan using: "explain pipeline ..... query ....."?
Will it print any indication that a projection was used?

**Comment by den-crane** [comment]
`allow_experimental_projection_optimization = 1` by default.

It marked obsolete by a mistake.

**Comment by arunmarathe** [comment]
> `allow_experimental_projection_optimization = 1` by default.
> 
> It marked obsolete by a mistake.
> 
> You can disable it `allow_experimental_projection_optimization = 0` if needed (if some queries produce incorrect results).

Thanks; I would still like to know how I would know if a projection was used by a query. I would expect it to appear in explain output, but for now, any other indication would be ok too.

**Comment by SaltTan** [comment]
> I would still like to know how I would know if a projection was used by a query.

select projections from system.query_log

**Comment by arunmarathe** [comment]
> > I would still like to know how I would know if a projection was used by a query.
> 
> select projections from system.query_log

Thanks!

**Comment by amosbird** [comment]
Closed by #49719
